### PR TITLE
Upgrade to Mutiny 0.17.0 and Smallrye Reactive Utils 2.5.1

### LINKED ISSUE #16430: Regression Test: Validation errors on Reactive routes are no longer propagated
## Describe the bug
Having these two routes:

```java
@Inject
FruitResource fruitResource;

@Route(methods = HttpMethod.POST, path = "/insert")
void insert(@Body Fruit fruit, RoutingContext context) {
    fruitResource.insertFruit(fruit)
            .onFailure().invoke(context::fail)
            .subscribe().with(id -> context.response().setStatusCode(201).end());
}

@Route(path = "/*", type = Route.HandlerType.FAILURE, produces = "application/json")
void databaseConstraintFailure(PgException e, HttpServerResponse response) {
    response.setStatusCode(400).end();
}
```

The `insert` route will insert a fruit item into the database. The `databaseConstraintFailure` handler will map the database constraints into a 400 Status code.

The above scenario works fine using 1.13.1.Final or later, but not using 999-SNAPSHOT which fails with the next exception:

```
ERROR: HTTP Request to /insert failed, error id: f1aab3c8-a7f2-464d-8c33-bd4a33af5e31-1
io.vertx.core.VertxException: Transaction already completed
	at io.vertx.sqlclient.impl.TransactionImpl.checkPending(TransactionImpl.java:115)
	at io.vertx.sqlclient.impl.TransactionImpl.rollback(TransactionImpl.java:154)
	at io.vertx.sqlclient.impl.TransactionImpl.rollback(TransactionImpl.java:159)
	at io.vertx.mutiny.sqlclient.Transaction.lambda$rollback$2(Transaction.java:112)
	at io.smallrye.mutiny.vertx.AsyncResultUni.subscribe(AsyncResultUni.java:30)
	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:37)
	at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni.subscribe(UniOnItemTransformToUni.java:25)
	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:37)
	at io.smallrye.mutiny.operators.uni.UniOnItemOrFailureFlatMap$UniOnItemOrFailureFlatMapProcessor.performInnerSubscription(UniOnItemOrFailureFlatMap.java:99)
	at io.smallrye.mutiny.operators.uni.UniOnItemOrFailureFlatMap$UniOnItemOrFailureFlatMapProcessor.onFailure(UniOnItemOrFailureFlatMap.java:65)
	at io.smallrye.mutiny.operators.uni.UniOperatorProcessor.onFailure(UniOperatorProcessor.java:44)
	at io.smallrye.mutiny.vertx.AsyncResultUni.lambda$subscribe$1(AsyncResultUni.java:36)
	at io.vertx.mutiny.sqlclient.PreparedQuery$2.handle(PreparedQuery.java:140)
	at io.vertx.mutiny.sqlclient.PreparedQuery$2.handle(PreparedQuery.java:135)
	at io.vertx.core.impl.future.FutureImpl$3.onFailure(FutureImpl.java:128)
	at io.vertx.core.impl.future.FutureBase.emitFailure(FutureBase.java:79)
	at io.vertx.core.impl.future.FutureImpl.tryFail(FutureImpl.java:198)
	at io.vertx.core.impl.future.PromiseImpl.tryFail(PromiseImpl.java:23)
	at io.vertx.sqlclient.impl.QueryResultBuilder.tryFail(QueryResultBuilder.java:118)
	at io.vertx.core.Promise.fail(Promise.java:89)
	at io.vertx.core.Promise.handle(Promise.java:53)
	at io.vertx.sqlclient.impl.TransactionImpl.lambda$null$0(TransactionImpl.java:75)
	at io.vertx.core.impl.future.FutureImpl$3.onSuccess(FutureImpl.java:124)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:62)
```

## To Reproduce
1.  git clone `https://github.com/Sgitario/quarkus-examples`
2.  cd quarkus-examples/reproducers/reactive-sql
3. mvn clean verify -Dtest=FruitResourceTest#shouldNotInsertFruitIfValidationError

The test should pass, but it's failing with the above exception

### Quarkus version or git rev
999-SNAPSHOT

### PR DESCRIPTION
**Pull request by jponge**

### CODE REVIEW

**Comment by jponge** [comment]
There's an issue with Kotlin and enforcer rules, so we can't check further.

@evanchooly any idea?

**Comment by gsmet** [comment]
```
Warning:  
12884
Dependency convergence error for org.jetbrains.kotlinx:kotlinx-coroutines-core:1.4.3 paths to dependency are:
12885
+-io.quarkus:quarkus-kotlin:999-SNAPSHOT
12886
  +-io.smallrye.reactive:mutiny-kotlin:0.17.0
12887
    +-org.jetbrains.kotlinx:kotlinx-coroutines-core:1.4.3
12888
and
12889
+-io.quarkus:quarkus-kotlin:999-SNAPSHOT
12890
  +-org.jetbrains.kotlinx:kotlinx-coroutines-jdk8:1.4.2
12891
    +-org.jetbrains.kotlinx:kotlinx-coroutines-core:1.4.2
```

**Comment by jponge** [comment]
Ok, should be fine to fix

File: extensions/keycloak-authorization/deployment/src/test/java/io/quarkus/keycloak/pep/test/ProtectedResource.java [hunk 1]
```diff
@@ -7,12 +7,7 @@
 
 import javax.inject.Inject;
 import javax.security.auth.AuthPermission;
-import javax.ws.rs.Consumes;
-import javax.ws.rs.ForbiddenException;
-import javax.ws.rs.GET;
-import javax.ws.rs.POST;
-import javax.ws.rs.Path;
-import javax.ws.rs.Produces;
+import javax.ws.rs.*;
```
Comment (geoand) [inline extensions/keycloak-authorization/deployment/src/test/java/io/quarkus/keycloak/pep/test/ProtectedResource.java:10]: Can you revert this change please?

File: integration-tests/resteasy-reactive-kotlin/prod-mode/pom.xml [hunk 2]
```diff
@@ -45,7 +45,7 @@
         <dependency>
             <groupId>org.jetbrains.kotlinx</groupId>
             <artifactId>kotlinx-coroutines-core-jvm</artifactId>
-            <version>1.4.2</version>
+            <version>${kotlinx-coroutines-core-jvm.version}</version>
```
Comment (geoand) [inline integration-tests/resteasy-reactive-kotlin/prod-mode/pom.xml:48]: I think this is now in the BOM, no?

**Comment by gsmet** [comment]
@jponge could you address @geoand 's comments above? Thanks!

File: integration-tests/resteasy-reactive-kotlin/prod-mode/pom.xml (hunk 2 above)
Comment (jponge) [reply integration-tests/resteasy-reactive-kotlin/prod-mode/pom.xml:48]: There's no Kotlin coroutines version property available from this project.

Comment (geoand) [reply integration-tests/resteasy-reactive-kotlin/prod-mode/pom.xml:48]: I see:

```xml
            <dependency>
                <groupId>org.jetbrains.kotlinx</groupId>
                <artifactId>kotlinx-coroutines-core-jvm</artifactId>
                <version>${kotlin.coroutine.version}</version>
            </dependency>
```

in the BOM, so the version can be dropped here

**Review by geoand — APPROVED** [review]

File: integration-tests/resteasy-reactive-kotlin/prod-mode/pom.xml (hunk 2 above)
Comment (geoand) [reply integration-tests/resteasy-reactive-kotlin/prod-mode/pom.xml]: Great, thanks!

**Comment by jponge** [comment]
@geoand and enforcer is complaining again, I'm trying to make it work locally

**Comment by geoand** [comment]
The enforcer requests that the deployment modules for tests match the runtime modules - this is needed to ensure the proper build  order for modules.

I didn't mention anything in the commit you removed the `deployment` modules because I thought they were left overs from a previous impl, but that is likely not the case. If I were you, I would simply add them back.

**Comment by jponge** [comment]
(testing CI, commits will be eventually squashed)

**Comment by geoand** [comment]
When you are done, please squash the intermediate commits

**Comment by jponge** [comment]
Hold on before anyone merges, I'm releasing Smallrye Reactive Utils 2.5.1 with a fix for 16430

**Comment by jponge** [comment]
https://github.com/smallrye/smallrye-reactive-utils/pull/313

**Review by gsmet — APPROVED** [review]

**Comment by jponge** [comment]
@evanchooly any objection to the Kotlin coroutines version 1.4.3 bump / alignment?

**Comment by evanchooly** [comment]
@jponge none at all!  (sorry for the late reply.  i saw the notification on my phone and then it disappeared in to the ether.)

**Review by evanchooly — APPROVED** [review]

**Comment by jponge** [comment]
One last CI run with a fix to remove `${mutiny-vertx.version}` redundancy against `${smallrye-reactive-utils.version}`
